specification.markdown

```diff
@@ -64,30 +64,67 @@ followed by the Data Size, and then the non-interpreted Binary itself:
-    size of the Element is unknown, which is a special case that we
-    believe will be useful for live streaming purposes. However, avoid
-    using this reserved word unnecessarily, because it makes parsing
-    slower and more difficult to implement.
+## Element Data Size
+
+The Element Data Size expresses the length in octets of Element Data.
+The Element Data Size itself MUST be encoded as a Variable Size Integer.
+By default, EBML Element Data Sizes can be encoded in lengths from one
+octet to eight octets, although Element Data Sizes of greater lengths
+MAY be used if the octet length of the EBML document's longest Element
+Data Size is declared in the EBMLMaxSizeLength Element of the EBML
+Header. Unlike the VINT\_DATA of the Element ID, the VINT\_DATA
+component of the Element Data Size is NOT REQUIRED to be encoded at the
+shortest valid length. For example, an Element Data Size with binary
+encoding of 1011 1111 or a binary encoding of 0100 0000 0011 1111 are
+both valid Element Data Sizes and both store a semantically equal value.
+
+Although an Element ID with all VINT\_DATA bits set to zero is invalid,
+an Element Data Size with all VINT\_DATA bits set to zero is allowed.
+This indicates that the Element Data of the Element is zero octets in
+length. Such an Element is referred to as an Empty Element.
+
+An Element Data Size with all VINT\_DATA bits set to one is reserved as
+an indicator that the size of the Element is unknown. This rule allows
+for an Element to be written and read before the size of the Element is
+known; however unknown Element Data Size values SHOULD NOT be used
+unnecessarily. An Element with an unknown Element Data Size MUST be a
+Master-element in that it contains other EBML Elements as sub-elements.
+The end of the Master-element is determined by the beginning of the next
+element that is not a valid sub-element of the Master-element. The use
+of Elements of unknown size is dependent on the definition of the EBML
+Schema declared in DocType, because an Element of unknown size can not
+be parsed without a complete list of all possible sub-elements.
+
+For Element Data Sizes encoded at octet lengths from one to eight, this
+table depicts the range of possible values that can be encoded as an
+Element Data Size. An Element Data Size with an octet length of 8 is
+able to express a size of 2^56-2 or 72,057,594,037,927,934 octets (or
+about 72 petabytes).
+
+Octet Length | Possible Value Range
+-------------|---------------------
+1            | 0 to  2^7-2
+2            | 0 to 2^14-2
+3            | 0 to 2^21-2
+4            | 0 to 2^28-2
+5            | 0 to 2^35-2
+6            | 0 to 2^42-2
+7            | 0 to 2^49-2
+8            | 0 to 2^56-2
+
+If the length of Element Data equals 2^(n*7)-1 then the octet length of
+the Element Data Size MUST be at least n+1. This rule prevents an
+Element Data Size from being expressed as a reserved value. For example,
```

Comment:
@robUx4 I changed
> An Element Data Size with all VINT\_DATA bits set to one is reserved as
> an indicator that the size of the Element is unknown. 

to

> An Element Data Size with all VINT\_DATA bits set to one is reserved as
an indicator that the size of the Element is unknown. The only reserved
value for the VINT\_DATA of Element Data Size is all bits set to one.